spaghetto/bin/src/Main.purs

```diff
@@ -411,6 +444,22 @@ main =
             runSpago buildEnv (Build.run options)
             testEnv <- runSpago env (mkTestEnv args)
             runSpago testEnv Test.run
+          LsPackages args -> do
+            let fetchArgs = { packages: mempty, selectedPackage: Nothing, ensureRanges: false }
+            { env, fetchOpts } <- mkFetchEnv fetchArgs
+            -- TODO: --no-fetch flag
+            dependencies <- runSpago env (Fetch.run fetchOpts)
```

Comment:
:question: Currently downloads packages. There was a comment regarding `no-fetch`, but maybe in this case (`ls packages`), this is the wrong way to gather `dependencies` to begin with? There is also a different comment, that all packages seem to be "local", and these could be related.

Reply:
The old spago could put together a build plan without downloading things, because all the dependencies had to be specified in the package set.
This is not the case anymore in the new spago, as we do not force users to type down all the dependencies beforehand, but we allow for them to be read from the config files contained in the remote repos.
This means that in the general case there's no way to put together a build plan before downloading all the dependencies, as some of them might be remote git repos.
It is possible in some specific cases, e.g. when all the dependencies of your project come from the registry. In that case we can look at the registry index for dependency information.

What this means for the above code is that I think we are good to proceed as is: it makes sense that the only way to get a build plan is to call `Fetch.run`, and we'll eventually introduce a `--no-fetch` or `--offline` flag to be handled by the fetching code (so this TODO can stay in place too)

Reply:
Question answered